config.go

```diff
@@ -341,6 +341,8 @@ type config struct {
 	Watchtower *lncfg.Watchtower `group:"watchtower" namespace:"watchtower"`
 
 	LegacyProtocol *lncfg.LegacyProtocol `group:"legacyprotocol" namespace:"legacyprotocol"`
+
+	AllowCircularRoute bool `long:"allow-circular" description:"If true, our node will allow htlc forwards that arrive and depart on the same channel."`
```

Comment:
Match var name and flag name? Is there a use case for allowing this btw?

Reply:
I think that keysend will replace the need for this eventually, but I'm worried about this change breaking no-invoice apps which tip by dropping off fees (could be the case that they drop off over the same channel for single channel nodes) in the meantime.